src/classify/trainingsampleset.cpp

```diff
@@ -618,6 +618,7 @@ void TrainingSampleSet::ComputeCanonicalSamples(const IntFeatureMap& map,
           if (dist > max_dist) {
             max_dist = dist;
             if (dist > max_max_dist) {
+              max_max_dist = dist;
               max_s1 = s1;
```

Comment:
That fixes a bug, thank you. Could you create a separate pull request for this fix?

Reply:
Sure, thanks @stweil